python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -616,21 +616,34 @@ def _from_node(
     @public
     @property
     def can_subset(self) -> bool:
+        """bool: If True, indicates that this AssetsDefinition may materialize any subset of its
+        asset keys in a given computation (as opposed to being required to materialize all asset
+        keys).
+        """
         return self._can_subset
 
     @public
     @property
     def group_names_by_key(self) -> Mapping[AssetKey, str]:
+        """Mapping[AssetKey, str]: Returns a mapping from the asset keys in this AssetsDefinition
```

Comment:
[1] This docstring suggests the value type here should be `Optional[str]`-- should it be?

Reply:
Yeah it's a bit weird (if there's no group name, then there's no entry in this dictionary)